File: src.csharp/AlphaTab/Core/EcmaScript/BigInt.cs
namespace AlphaTab.Core.EcmaScript;

internal static class BigInt
{
    public static long AsUintN(int bits, long value)
    {
        // https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-bigint.asuintn
        return (long)(value % Math.Pow(2, bits));
    }
}

File: packages/kotlin/src/android/src/main/java/alphaTab/core/ecmaScript/BigInt.kt
package alphaTab.core.ecmaScript

internal class BigInt {
    companion object {
        fun asUintN(bits: Double, value: Long): Long {
            // https://tc39.es/ecma262/multipage/numbers-and-dates.html#sec-bigint.asuintn
            return (value % Math.pow(2.0, bits)).toLong()
        }
    }
}
